### ISSUE TITLE
Error IDA 7

### ISSUE THREAD

**Issue by icspe** [x64dbg/x64dbgida#4]
Hi, The plugin fails to start in IDA Version 7. Works perfectly in v6.95.
"
Failed while executing plugin_t.init():
Traceback (most recent call last):
  File "C:/Program Files/IDA 7.0/plugins/x64dbgida.py", line 206, in init
    menu = idaapi.add_menu_item("Edit/x64dbgida/", "About", "", 0,
AttributeError: 'module' object has no attribute 'add_menu_item'
"

**Comment by mrexodia** [comment]
If you send me an ida copy to test with I will fix it for you...

**Comment by icspe** [comment]
I can't do that, but I'll try and fix it and share the results if I'm lucky.

**Comment by decidedlygray** [comment]
@icspe FWIW: This is due to the fact that IDA 7.0 is enforcing the use of the Actions API. It was introduced in IDA 6.7, but still allowed plugin writers to use the old APIs. With the release of 7.0, those old APIs have been fully removed.

Additional details for @mrexodia if this makes his list of things he wants to work on (though I suspect his plate is always full), or to any contributors willing to work on this: http://www.hexblog.com/?p=886

Basically, since IDA 6.7 you aren't supposed to be using the old API `add_menu_item`, and are instead supposed to define a handler class then properly register it. That blog post has example code on how this is accomplished.

**Comment by mrexodia** [comment]
@icspe please try out https://github.com/techbliss/x64dbgida/blob/21cf0d6403c82dca0157b534f5df99370e0de756/x64dbgida.py and report if it's working, then we can merge #5

**Comment by pmenke968** [comment]
@mrexodia Thanks for this.  Worked for me on 7.0.170914 Windows x64.

**Comment by techbliss** [comment]
thank you @pmenke968 for testing it.
My self have tested it with 6.6, 6.8, 6.95, and 7.0 and should not have any conflicts.
